library/src/test/java/com/yayandroid/locationmanager/providers/locationprovider/DispatcherLocationProviderTest.java

```diff
@@ -295,6 +299,80 @@ public void resolveGooglePlayServicesShouldContinueWithDefaultWhenResolveDialogI
         verify(dispatcherLocationProvider).continueWithDefaultProviders();
     }
 
+    @Test
+    public void resolveGooglePlayServicesShouldContinueWithDefaultWhenErrorCannotBeResolved() {
```

Comment:
Why is the following not enough?

```
int unresolvableError = ConnectionResult.SERVICE_INVALID;

when(dispatcherLocationSource.getGoogleApiErrorDialog(
      eq(activity), eq(unresolvableError),  eq(RequestCode.GOOGLE_PLAY_SERVICES), any(OnCancelListener.class))
).thenReturn(dialog);

dispatcherLocationProvider.resolveGooglePlayServices(unresolvableError);
verify(dialog).show();

dialog.dismiss(); // Simulate dismiss dialog (error cannot be resolved)

verify(dispatcherLocationProvider).continueWithDefaultProviders();
```

What are you trying to test by catching listeners?

Reply:
To check that `continueWithDefaultProviders()` is called when error is unresolvable and dialog is dismissed (pressed OK) but not cancelled.
`continueWithDefaultProviders()` called in OnDismissListener, so to simulate real listener call, I catch and store it. 
The `dialog` is `@Mock` object and `dialog.dismiss()` does nothing.